metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/entities/EntitiesController.java

```diff
@@ -103,21 +108,28 @@ public ResponseEntity<UrnResponseMap> getEntities(
 
   @PostMapping(value = "/", produces = MediaType.APPLICATION_JSON_VALUE)
   public ResponseEntity<List<String>> postEntities(
-      @RequestBody @Nonnull List<UpsertAspectRequest> aspectRequests) {
+      @RequestBody @Nonnull List<UpsertAspectRequest> aspectRequests,
+      @RequestHeader(name = Constants.IF_UNMODIFIED_SINCE) String condUpdate) {
     log.info("INGEST PROPOSAL proposal: {}", aspectRequests);
 
-    Authentication authentication = AuthenticationContext.getAuthentication();
-    String actorUrnStr = authentication.getActor().toUrnStr();
-
-    List<Pair<String, Boolean>> responses = aspectRequests.stream()
-        .map(MappingUtil::mapToProposal)
-        .map(proposal -> MappingUtil.ingestProposal(proposal, actorUrnStr,  _entityService, _objectMapper))
-        .collect(Collectors.toList());
-    if (responses.stream().anyMatch(Pair::getSecond)) {
-      return ResponseEntity.status(HttpStatus.CREATED)
-          .body(responses.stream().filter(Pair::getSecond).map(Pair::getFirst).collect(Collectors.toList()));
-    } else {
-      return ResponseEntity.ok(Collections.emptyList());
+    try {
+      Authentication authentication = AuthenticationContext.getAuthentication();
+      String actorUrnStr = authentication.getActor().toUrnStr();
+      Map<String, Long> createdOnMap = CondUpdateUtil.extractCondUpdate(condUpdate);
+      List<Pair<String, Boolean>> responses = aspectRequests.stream()
+              .map(MappingUtil::mapToProposal)
+              .map(proposal -> MappingUtil.ingestProposal(proposal, actorUrnStr, _entityService, _objectMapper,
+                      createdOnMap.get(actorUrnStr + "+" + proposal.getAspectName())))
+              .collect(Collectors.toList());
+      if (responses.stream().anyMatch(Pair::getSecond)) {
+        return ResponseEntity.status(HttpStatus.CREATED)
+                .body(responses.stream().filter(Pair::getSecond).map(Pair::getFirst).collect(Collectors.toList()));
+      } else {
+        return ResponseEntity.ok(Collections.emptyList());
+      }
+    } catch (PreconditionFailedException e) {
```

Comment:
This can be pushed into the GlobalControllerExceptionHandler class to avoid wrapping both methods